magento/graphql-ce
# Fix to resolve an issue with Date, Time, Date&Time in customizable_options

### Description (*)

This PR allows to send Time, Date, Date&Time

### Fixed Issues (if relevant)

1. magento/graphql-ce#345: Type CustomizableOptionInput supports only String! for value

### Manual testing scenarios (*)

1. ...
2. Check that something similar can be achieved  
```graphql
{
  "data": {
    "addSimpleProductsToCart": {
      "cart": {
        "items": [
          {
            "product": {
              "id": 1,
              "name": "Simple Product",
              "sku": "simple-product"
            },
            "qty": 9,
            "customizable_options": [
              {
                "id": 1,
                "label": "Date Test",
                "is_required": 1,
                "type": "date",
                "values": [
                  {
                    "price": {
                      "type": "FIXED",
                      "units": "$",
                      "value": 10
                    },
                    "value": "Feb 12, 2019"
                  }
                ]
              },
              {
                "id": 2,
                "label": "Time Test",
                "is_required": 1,
                "type": "time",
                "values": [
                  {
                    "price": {
                      "type": "FIXED",
                      "units": "$",
                      "value": 0.5
                    },
                    "value": "1:45 AM"
                  }
                ]
              },
              {
                "id": 3,
                "label": "Date And Time Test",
                "is_required": 1,
                "type": "date_time",
                "values": [
                  {
                    "price": {
                      "type": "FIXED",
                      "units": "$",
                      "value": 1.5
                    },
                    "value": "3/21/19, 8:00 AM"
                  }
                ]
              }
            ]
          }
        ]
      }
    }
  }
}
```

by executing

```graphql
mutation {
  addSimpleProductsToCart(
    input: {
      cart_id: "b1aNSLjkaBKB1UfTrlaR38OB2vvsNRyw"
      cartItems: [
        {
          data: {
            qty: 1
            sku: "simple-product"
          }
          customizable_options: [
            {
              id: 1
              value: "2019-02-12 03:45:00"
            }
            {
              id: 2
              value: "2019-04-17 13:45:00"
            }
            {
              id: 3
              value: "2019-03-20 20:00:00"
            }
          ]
        }
      ]
    }
  ) {
    cart {
      items {
        product {
          id
          name
          sku
        }
        qty
        ... on SimpleCartItem {
          customizable_options {
            id
            label
            is_required
            type
            values {
              price {
                type
                units
                value
              }
              value
            }
          }
        }
      }
    }
  }
}
```


### Contribution checklist (*)
